File: templates/cocos2d/TwoScene/Assets/MenuScene.js
//
//  Created using [RapidGame](http://wizardfu.com/rapidgame).
//  See the `LICENSE` file for the license governing this code.
//  Developed by Nat Weiss.
//

var MenuScene = cc.Scene.extend({
	layer: null,
	
	onEnter: function() {
		this._super();
		this.layer = new MenuLayer();
		this.layer.init();
		this.addChild(this.layer);
	}
});

var MenuLayer = cc.Layer.extend({
	bg: null,
	bg2: null,
	logo: null,
	logoLabel: null,
	menu: null,
	playButton: null,
	exitButton: null,
	
	init: function() {
		var self = this,
			font = Game.config["font"],
			logoText,
			x, y;

		this._super();

		logoText = "TwoScene";
		Game.playEffect("Intro.mp3");

		// Background
		this.bg = cc.LayerColor.create(cc.color(218, 214, 212, 255));
		this.addChild(this.bg, -1);
		
		// Logo
		y = 24;
		this.logo = cc.Sprite.create("Logo.png");
		this.logo.setPosition(Game.centralize(0, 228));
		this.addChild(this.logo, 1);
		this.logo.y -= y * .5;
		this.logo.runAction(cc.RepeatForever.create(cc.Sequence.create(
			cc.EaseInOut.create(cc.MoveBy.create(2, cc.p(0, y)), 1.2),
			cc.EaseInOut.create(cc.MoveBy.create(2, cc.p(0, -y)), 1.2)
		)));

		// Title
		this.logoLabel = cc.LabelTTF.create(
			Game.getLocalizedString("title"),
			font,
			200
		);
		this.logoLabel.setColor(cc.color(128, 128, 128));
		this.logoLabel.setPosition(Game.centralize(0, 228));
		this.addChild(this.logoLabel, 1);

		// Menu
		this.menu = cc.Menu.create();
		this.menu.setPosition(cc.p());
		this.addChild(this.menu, 1);

		// Buttons
		this.playLabel = cc.MenuItemFont.create("Play", this.onPlayButton, this);
		this.playLabel.setPosition(Game.centralize(0, -400));
		this.playLabel.setFontSize(120);
		this.playLabel.setFontName(font);
		this.playLabel.setColor(cc.color(196, 196, 196));
		this.menu.addChild(this.playLabel);
		
		return true;
	},
	
	onPlayButton: function() {
		var scene = new GameScene;
		scene.init();
		cc.director.runScene(scene);
	}

});

File: templates/cocos2dx/TwoScene/Projects/MenuScene.cpp
///
/// > Created using [RapidGame](https://github.com/natweiss/rapidgame). See the `LICENSE` file for the license governing this code.
///

#include "Game.h"
#include "MenuScene.h"
#include "GameScene.h"

void MenuScene::onEnter()
{
	Scene::onEnter();

	string logoText = "TwoScene";
	string font = "DolceVita.ttf";
	CocosDenshion::SimpleAudioEngine::getInstance()->playBackgroundMusic("Intro.mp3");

	// Background
	auto bg = cocos2d::LayerColor::create(cocos2d::Color4B(218, 214, 212, 255));
	this->addChild(bg, -1);

	// Logo
	float y = 24;
	auto logo = cocos2d::Sprite::create("Logo.png");
	logo->setPosition(Game::centralize(0, 228 - y * 0.5f));
	this->addChild(logo, 1);
	logo->runAction(cocos2d::RepeatForever::create(cocos2d::Sequence::create(
		cocos2d::EaseInOut::create(cocos2d::MoveBy::create(2, cocos2d::Vec2(0, y)), 1.2),
		cocos2d::EaseInOut::create(cocos2d::MoveBy::create(2, cocos2d::Vec2(0, -y)), 1.2),
		nullptr
	)));
	
	// Title
	auto logoLabel = cocos2d::Label::createWithTTF(logoText.c_str(), font, 200);
	logoLabel->setColor(cocos2d::Color3B(128, 128, 128));
	logoLabel->setPosition(Game::centralize(0, 228));
	this->addChild(logoLabel, 1);
	
	// Menu
	auto menu = cocos2d::Menu::create();
	menu->setPosition(cocos2d::Vec2::ZERO);
	this->addChild(menu, 1);
	
	// Buttons
	auto playLabel = cocos2d::Label::createWithTTF("Play", font, 120);
	playLabel->setColor(cocos2d::Color3B(196, 196, 196));
	
	auto playButton = cocos2d::MenuItemLabel::create(playLabel, [] (Ref* ref)
	{
		auto scene = new GameScene;
		cocos2d::Director::getInstance()->replaceScene(scene);
		scene->release();		
	});
	playButton->setPosition(Game::centralize(0, -400));
	menu->addChild(playButton);
}


